libc/src/__support/GPU/utils.h

```diff
@@ -31,6 +31,25 @@ LIBC_INLINE bool is_first_lane(uint64_t lane_mask) {
   return gpu::get_lane_id() == get_first_lane_id(lane_mask);
 }
 
+/// Gets the sum of all lanes inside the warp or wavefront.
```

Comment:
We have llvm.amdgcn.wave.reduce.umin/umax. The intent was to expand those intrinsics to cover more operations , since the expansion has different strategies depending on the available instructions

Reply:
I'll need this for the NVPTX build in any case, and I think it's sufficient for now to have this in source the way I need it until someone wants to implement the suite more correctly. However I don't think it will change too much considering that this is just convergent addition.

Reply:
Also looks like the alternative implementations were never implemented. currently looks like we don't have the bpermute or DPP paths

Reply:
I definitely see the appeal of having these as intrinsic functions. Specifically because right now I need to use `__builtin_amdgcn_wavefrontsize` which prevents these loops from being fully unrolled. However, I think it will be easy to drop those intrinsic functions in later.